### PR TITLE
add mypy and update files

### PR DESCRIPTION
**Pull request by dbasunag**

### CODE REVIEW

**Comment by msi-b1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: apps/polarion/polarion_utils.py [hunk 1]
```diff
@@ -2,31 +2,29 @@
...
 
 LOGGER = get_logger(name=__name__)
 
 
-def git_diff():
+def git_diff() -> Any:
     data = subprocess.check_output(shlex.split("git diff HEAD^-1"))
-    data = data.decode("utf-8")
-    return data
+    return data.decode("utf-8")
```
Comment (rnetser) [inline apps/polarion/polarion_utils.py:12]: `utf-8` is the default; can be removed

File: apps/polarion/polarion_utils.py [hunk 2]
```diff
@@ -2,31 +2,29 @@
...
 
 
-def git_diff_lines():
-    diff = {}
+def git_diff_lines() -> dict:
```
Comment (rnetser) [inline apps/polarion/polarion_utils.py:15]:
```suggestion
def git_diff_lines() -> Dict:
```

File: apps/polarion/polarion_utils.py [hunk 3]
```diff
@@ -2,31 +2,29 @@
...
             diff.setdefault("added", []).append(line)
-
     return diff
 
 
 def validate_polarion_requirements(
-    polarion_test_ids,
-    polarion_project_id,
-):
+    polarion_test_ids: list,
```
Comment (rnetser) [inline apps/polarion/polarion_utils.py:25]:
```suggestion
    polarion_test_ids: List,
```

File: apps/polarion/polarion_utils.py [hunk 4]
```diff
@@ -2,31 +2,29 @@
...
+    polarion_project_id: str,
+) -> list:
     from pylero.work_item import TestCase, Requirement
 
     tests_with_missing_requirements = []
```
Comment (rnetser) [inline apps/polarion/polarion_utils.py:30]:
```suggestion
    tests_with_missing_requirements : List = []
```

Comment (rnetser) [reply apps/polarion/polarion_utils.py:30]: typing is missing in some definitions

File: apps/polarion/polarion_verify_tc_requirements.py [hunk 5]
```diff
@@ -21,7 +22,7 @@
 )
 @click.option("--project-id", "-p", help="Provide the polarion project id")
 @click.option("--verbosity", default=False, is_flag=True)
-def has_verify(config_file_path, project_id, verbosity):
+def has_verify(config_file_path: Any, project_id: str, verbosity: bool) -> Any:
```
Comment (rnetser) [inline apps/polarion/polarion_verify_tc_requirements.py:25]: why `config_file_path` is `Any`?

Comment (rnetser) [reply apps/polarion/polarion_verify_tc_requirements.py:25]: why `-> Any`, the function does not return anything

File: apps/unused_code/unused_code.py [hunk 6]
```diff
@@ -23,15 +24,15 @@ def is_fixture_autouse(func):
...
 
 
-def _iter_functions(tree):
+def _iter_functions(tree: Any) -> Any:
     """
     Get all function from python file
     """
 
-    def is_func(_elm):
+    def is_func(_elm: Any) -> Any:
```
Comment (rnetser) [inline apps/unused_code/unused_code.py:32]: why Any? please fix in all places. this function returns bool

**Review by rnetser — CHANGES_REQUESTED** [review]

**Pull request merged**